src/chart/components/pane/pane.component.ts

```diff
@@ -108,9 +107,10 @@ export class PaneComponent extends ChartBaseElement {
 			this.canvasBoundsContainer
 				.observeBoundsChanged(CanvasElement.PANE_UUID(this.uuid))
 				.pipe(distinctUntilChanged(areBoundsChanged))
-				.subscribe(() => {
+				.subscribe(bounds => {
 					this.yExtentComponents.forEach(c => c.scale.recalculateZoomY());
 					this.dynamicObjectsCanvasModel.fireDraw();
+					bounds.height <= 0 ? this.hide() : this.show();
```

Comment:
Looks like a circular dependency: `hide()/show() => recalculateBounds`
In this case,` bounds changed => hide()/show()`
How is it possible?

Reply:
I'm taking about responsibility principle. Who is responsible for calling hide/show? We change pane's state (hide/show) => bounds recalculated. It's good.
But why we need to change pane's state on bounds change? bounds is just a model, that calculates panes ratios based on provided panes and their order, it cannot control pane's state